src/Operators/vector_rotation_operators.jl

```diff
@@ -0,0 +1,127 @@
+
+_extrinsic_ reference frames are:
+
+- Cartesian for box domains 
+- Latitude - Longitude for spherical domains
+
+Therefore, for the `RectilinearGrid` and the `LatitudeLongitudeGrid`, the _extrinsic_ and the 
+_intrinsic_ reference frames are equivalent
+"""
+@inline intrinsic_vector(i, j, k, grid::AbstractGrid, uₑ, vₑ, wₑ) = 
+    getvalue(uₑ, i, j, k, grid), getvalue(vₑ, i, j, k, grid), getvalue(wₑ, i, j, k, grid)
+
+"""
+    extrinsic_vector(i, j, k, grid::AbstractGrid, uᵢ, vᵢ, wᵢ)
+
+Convert the three-dimensional vector with components `uₑ, vₑ, wₑ` defined on the reference 
+reference frame _intrinsic_ to the grid, to the _extrinsic_ reference frame associated with the domain.
+
+_extrinsic_ reference frames are:
+
+- Cartesian for box domains 
+- Latitude - Longitude for spherical domains
+
+Therefore, for the `RectilinearGrid` and the `LatitudeLongitudeGrid`, the _extrinsic_ and the 
+_intrinsic_ reference frames are equivalent
+"""
+@inline extrinsic_vector(i, j, k, grid::AbstractGrid, uᵢ, vᵢ, wᵢ) =
+    getvalue(uᵢ, i, j, k, grid), getvalue(vᵢ, i, j, k, grid), getvalue(wᵢ, i, j, k, grid)
+
+
+
+# Intrinsic and extrinsic conversion for `OrthogonalSphericalShellGrid`s,
+# i.e. curvilinear grids defined on a sphere which are locally orthogonal.
+# If the coordinates match with the coordinates of a latitude-longitude grid
+# (i.e. globally orthogonal), these functions collapse to 
+# uₑ, vₑ, wₑ = uᵢ, vᵢ, wᵢ
+@inline function intrinsic_vector(i, j, k, grid::OrthogonalSphericalShellGrid, uᵢ, vᵢ, wᵢ)
+
+    φᶜᶠᵃ₊ = φnode(i, j+1, 1, grid, Center(), Face(), Center())
+    φᶜᶠᵃ₋ = φnode(i,   j, 1, grid, Center(), Face(), Center())
+    Δyᶜᶜᵃ = Δyᶜᶜᶜ(i,   j, 1, grid)
+
+    ũ = deg2rad(φᶜᶠᵃ₊ - φᶜᶠᵃ₋) / Δyᶜᶜᵃ
+
+    φᶠᶜᵃ₊ = φnode(i+1, j, 1, grid, Face(), Center(), Center())
+    φᶠᶜᵃ₋ = φnode(i,   j, 1, grid, Face(), Center(), Center())
+    Δxᶜᶜᵃ = Δxᶜᶜᶜ(i,   j, 1, grid)
+
+    ṽ = - deg2rad(φᶠᶜᵃ₊ - φᶠᶜᵃ₋) / Δxᶜᶜᵃ
+
+    𝒰 = sqrt(ũ^2 + ṽ^2)
+
+    u  = getvalue(uₑ, i, j, k, grid)
+    v  = getvalue(vₑ, i, j, k, grid)
+    wᵢ = getvalue(wₑ, i, j, k, grid)
+
+    d₁ = ũ / 𝒰
+    d₂ = ṽ / 𝒰
+
+    uᵢ = u * d₁ + v * d₂
+    vᵢ = u * d₂ - v * d₁
+
+    return uᵢ, vᵢ, wᵢ
+end
+
+@inline function extrinsic_vector(i, j, k, grid::OrthogonalSphericalShellGrid, uᵢ, vᵢ, wᵢ)
+
+    φᶜᶠᵃ₊ = φnode(i, j+1, 1, grid, Center(), Face(), Center())
+    φᶜᶠᵃ₋ = φnode(i,   j, 1, grid, Center(), Face(), Center())
+    Δyᶜᶜᵃ = Δyᶜᶜᶜ(i,   j, 1, grid)
+
+    ũ = deg2rad(φᶜᶠᵃ₊ - φᶜᶠᵃ₋) / Δyᶜᶜᵃ
+
+    φᶠᶜᵃ₊ = φnode(i+1, j, 1, grid, Face(), Center(), Center())
+    φᶠᶜᵃ₋ = φnode(i,   j, 1, grid, Face(), Center(), Center())
+    Δxᶜᶜᵃ = Δxᶜᶜᶜ(i,   j, 1, grid)
+
+    ṽ = - deg2rad(φᶠᶜᵃ₊ - φᶠᶜᵃ₋) / Δxᶜᶜᵃ
+
+    𝒰 = sqrt(ũ^2 + ṽ^2)
+
+    u  = getvalue(uᵢ, i, j, k, grid)
+    v  = getvalue(vᵢ, i, j, k, grid)
+    wₑ = getvalue(wᵢ, i, j, k, grid)
+
+    d₁ = ũ / 𝒰
+    d₂ = ṽ / 𝒰
+
+    uₑ = u * d₁ - v * d₂
+    vₑ = u * d₂ + v * d₁
```

Comment:
same remark as `intrinsic_vector` method

Reply:
changed names as the remark above